zerver/data_import/import_util.py

```diff
@@ -699,15 +700,45 @@ def build_realm_emoji(realm_id: int, name: str, id: int, file_name: str) -> Zerv
     )
 
 
-def get_emojis(emoji_dir: str, emoji_url: str, emoji_path: str) -> str | None:
-    upload_emoji_path = os.path.join(emoji_dir, emoji_path)
+def get_emojis(
+    emoji_dir: str,
+    emoji_url: str,
+    emoji_name: str,
+    emoji_id: int,
+    realm_id: int,
+) -> str | None:
+    """
+    This downloads the given `emoji_url` into the given `emoji_dir`
+    and returns the file name of the downloaded emoji.
+
+    Returns `None` when the emoji file has an unspecified content type.
 
+    Raises `BadImageError` when the emoji's content type is not in
+    `THUMBNAIL_ACCEPT_IMAGE_TYPES` or `INLINE_MIME_TYPES`.
+    """
     response = requests.get(emoji_url, stream=True)
+    content_type = response.headers.get("Content-Type")
+    if content_type is None:
+        logging.warning(
+            "Emoji %s has an unspecified content type. Using the original file extension.",
+            emoji_name,
+        )
+        return None
```

Comment:
This `return None` seems at odds with the warning that implies it's continuing to actually download the image?

Reply:
Looks like this `if` branch was also lacking a testcase.